plugins/flytekit-airflow/flytekitplugins/airflow/task.py

```diff
@@ -1,51 +1,171 @@
+    The airflow task module, name and parameters are stored in the task config.
+
+    Some of the Airflow operators are not deferrable, For example, BeamRunJavaPipelineOperator, BeamRunPythonPipelineOperator.
+    These tasks don't have async method to get the job status, so cannot be used in the Flyte agent. We run these tasks in a container.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        task_config: AirflowObj,
+        inputs: Optional[Dict[str, Type]] = None,
+        **kwargs,
+    ):
+        super().__init__(
+            name=name,
+            task_config=task_config,
+            interface=Interface(inputs=inputs or {}),
+            **kwargs,
+        )
+        self._task_resolver = airflow_task_resolver
+
+    def execute(self, **kwargs) -> Any:
+        logger.info("Executing Airflow task")
+        _get_airflow_instance(self.task_config).execute(context=Context())
+
+
+class AirflowTask(AsyncAgentExecutorMixin, PythonTask[AirflowObj]):
+    """
+    This python task is used to wrap an Airflow task. It is used to run an Airflow task in Flyte agent.
+    The airflow task module, name and parameters are stored in the task config. We run the Airflow task in the agent.
+    """
+
     _TASK_TYPE = "airflow"
 
     def __init__(
         self,
         name: str,
-        query_template: str,
-        task_config: Optional[AirflowConfig],
+        task_config: Optional[AirflowObj],
         inputs: Optional[Dict[str, Type]] = None,
         **kwargs,
     ):
         super().__init__(
             name=name,
             task_config=task_config,
-            query_template=query_template,
             interface=Interface(inputs=inputs or {}),
             task_type=self._TASK_TYPE,
             **kwargs,
         )
 
     def get_custom(self, settings: SerializationSettings) -> Dict[str, Any]:
+        # Use jsonpickle to serialize the Airflow task config since the return value should be json serializable.
         return {"task_config_pkl": jsonpickle.encode(self.task_config)}
 
 
+def _get_airflow_instance(airflow_obj: AirflowObj) -> typing.Union[BaseOperator, BaseSensorOperator, BaseTrigger]:
+    # Set the GET_ORIGINAL_TASK attribute to True so that obj_def will return the original
+    # airflow task instead of the Flyte task.
+    ctx = FlyteContextManager.current_context()
+    ctx.user_space_params.builder().add_attr("GET_ORIGINAL_TASK", True).build()
+
+    obj_module = importlib.import_module(name=airflow_obj.module)
+    obj_def = getattr(obj_module, airflow_obj.name)
+    if issubclass(obj_def, BaseOperator) and not issubclass(obj_def, BaseSensorOperator) and _is_deferrable(obj_def):
+        try:
+            return obj_def(**airflow_obj.parameters, deferrable=True)
+        except AirflowException:
+            logger.debug(f"Airflow operator {airflow_obj.name} does not support deferring")
```

Comment:
do we have to return here?

Reply:
yes, we return at line 151 without `deferrable=True`

Reply:
got it. 

Another question: is there a more specific exception related to operators not being deferrable? Capturing `AirflowException` means that other exceptions are also treated as that, right? If that's not the case, can we get the exception message in the logs as well?

Reply:
It always raises AirflowException
https://github.com/apache/airflow/blob/3a084b70c2161a02a0cb0db39bd06dfb120de46d/airflow/models/baseoperator.py#L808-L811

Reply:
updated error message.